src/lib.rs

```diff
@@ -300,90 +301,39 @@ pub mod root_span_macro;
-compile_error!("feature \"opentelemetry_0_15\" and feature \"opentelemetry_0_16\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_15", feature = "opentelemetry_0_17"))]
-compile_error!("feature \"opentelemetry_0_15\" and feature \"opentelemetry_0_17\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_15", feature = "opentelemetry_0_18"))]
-compile_error!("feature \"opentelemetry_0_15\" and feature \"opentelemetry_0_18\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_15", feature = "opentelemetry_0_19"))]
-compile_error!("feature \"opentelemetry_0_15\" and feature \"opentelemetry_0_19\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_15", feature = "opentelemetry_0_20"))]
-compile_error!("feature \"opentelemetry_0_15\" and feature \"opentelemetry_0_20\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_16", feature = "opentelemetry_0_17"))]
-compile_error!("feature \"opentelemetry_0_16\" and feature \"opentelemetry_0_17\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_16", feature = "opentelemetry_0_18"))]
-compile_error!("feature \"opentelemetry_0_16\" and feature \"opentelemetry_0_18\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_16", feature = "opentelemetry_0_19"))]
-compile_error!("feature \"opentelemetry_0_16\" and feature \"opentelemetry_0_19\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_16", feature = "opentelemetry_0_20"))]
-compile_error!("feature \"opentelemetry_0_16\" and feature \"opentelemetry_0_20\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_17", feature = "opentelemetry_0_18"))]
-compile_error!("feature \"opentelemetry_0_17\" and feature \"opentelemetry_0_18\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_17", feature = "opentelemetry_0_19"))]
-compile_error!("feature \"opentelemetry_0_17\" and feature \"opentelemetry_0_19\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_17", feature = "opentelemetry_0_20"))]
-compile_error!("feature \"opentelemetry_0_17\" and feature \"opentelemetry_0_20\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_18", feature = "opentelemetry_0_19"))]
-compile_error!("feature \"opentelemetry_0_18\" and feature \"opentelemetry_0_19\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_18", feature = "opentelemetry_0_20"))]
-compile_error!("feature \"opentelemetry_0_18\" and feature \"opentelemetry_0_20\" cannot be enabled at the same time");
-
-#[cfg(all(feature = "opentelemetry_0_19", feature = "opentelemetry_0_20"))]
-compile_error!("feature \"opentelemetry_0_19\" and feature \"opentelemetry_0_20\" cannot be enabled at the same time");
+const _OTEL_GUARD: () = assert!({
```

Comment:
It breaks it.
Now it's runtime and not compile time.
I don't see the error while `build`ing

Reply:
I think it's because compiler fails earlier than running/replacing consts.

Reply:
No, it's not runtime, `const _: ()` forces compile-time evaluation of a const block here.

It might be a later compiler pass than checking import name clashes, so errors look different, but this will prevent compilation from working.

Reply:
Anyway, compilation will fail, even if you remove this bit.
Having `assert` is as useful as not having it. But `compile_error` will be triggered earlier.

Reply:
Okay, I can revert this if that's your preference, however I don't think maintaining an ever-expanding and quite a bit error prone Cartesian cross-product of possible flag combinations is a good approach.

Ideally, there could be some macro that would take a feature list and then construct the cross product for us, but I'm unaware of such a macro and then whether it's at all possible as `cfg!` and `#[cfg(..)]` expression are a bit magical and might be evaluated earlier than any possible macro could kick in.

Reply:
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=6a803585c7837deaf6f6cab6e4307533

Here you see double `a` inclusion, and `compile_error` but nothing about `assert`.
If you comment `compile_error` out. still you cannot see anything about `assert`.
If the code can compile, and there are no errors (which is not the case for this package), you can see `assert` shouting.

Reply:
I'm just nobody, I'm not the package maintainer :D

It's funny that before your last commit, it annoyed me a lot also and I thought about it :)) so I tried to write a macro like what you said. (I couldn't find an existing one)
I finished it yesterday... preparing the documentation and tests and CI of it

Reply:
If @LukeMathWalker agrees, we can use https://crates.io/crates/mutually_exclusive_features

Reply:
Happy to add the dependency!

Reply:
> If @LukeMathWalker agrees, we can use https://crates.io/crates/mutually_exclusive_features

Cool, done. Is the `@recurs [params: ...]` a common convention for recursive declarative macros BTW? Haven't seen it yet.

Reply:
I'm not sure how common it is. I learned that from `diesel` macros, [eg](https://github.com/diesel-rs/diesel/blob/b51567f6f4bf32d03c3c00b03f38b0a3c29899f0/diesel/src/type_impls/tuples.rs#L480).

It's not just for recursion, they can there to make it rarely possible to being called from outside, and make internal calls in macros.

Like [here](https://github.com/omid/mutually_exclusive_features/blob/master/src/lib.rs#L91C11-L91C11), I used it, to name and extract a functionality of making comma separated list of items, with the `comma_sep` name. (And yes, in this case we also have a recursion and an extra rule for the last element.)

And about the new changes, I think we need exactly one of the `opentelemetry_*` features, no? I'm not sure. If so, then better to use `exactly_one_of` macro.

Reply:
It's also written [here](https://veykril.github.io/tlborm/decl-macros/patterns/internal-rules.html) and [here](https://danielkeep.github.io/tlborm/book/pat-internal-rules.html) and [railroad](https://github.com/lukaslueg/macro_railroad) also flags them as `internal rules`
![image](https://github.com/LukeMathWalker/tracing-actix-web/assets/45714/6d475381-49ac-4c67-a2e7-641522a047d1)
vs 
![image](https://github.com/LukeMathWalker/tracing-actix-web/assets/45714/c2146d43-8cf4-4483-a3af-4c4fb5a7e389)